.github/workflows/release.yml

```
@@ -11,26 +11,31 @@ on:
 jobs:
   build:
     if: github.repository == 'jazzband/pip-tools'
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
 
-      - name: Cache
-        uses: actions/cache@v2
-        with:
-          path: ~/.cache/pip
-          key: deploy-${{ hashFiles('**/setup.py') }}
-          restore-keys: |
-            deploy-
-
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
           python-version: 3.9
 
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: release-${{ hashFiles('**/setup.py') }}
```

```suggestion
          key: release-${{ hashFiles('**/release.yml') }}
```

Another thing: we want to hash the filename of whatever defines the dependencies installed by pip. This could be `setup.py`, `setup.cfg`, `requirements.txt`, or a combination.

In this case, the stuff pip installs is defined in this file.

The idea is, when a dependency is added or removed, that file changes, and we get a new cache.

---

A bit more info about `actions/cache`:

First it looks for a cache matching the `key`. If found, great, it downloads it.

If not, it then it looks for any _partially_ matching `restore-keys`. (This is a list, even though we probably only need a single entry.)

So if no `key` was found (e.g. when we've added a new dependency), it'll download the old cache because of the partial `restore-keys` match. We then install our new stuff into the pip cache, and it'll upload that for use next time.

Docs: https://github.com/actions/cache#creating-a-cache-key

---

I've removed the caching bits for now, given the small time window in which it would be useful for a release workflow.